src/common/views.py

```diff
@@ -46,15 +52,47 @@ def common_list_view(request):
         with open(metadata_file) as file:
             metadata = json.load(file)
             try:
-                context['release_version'] = metadata['release_version']
+                current_app_version = metadata['release_version']
+                return current_app_version
             
             except KeyError:
-                context['release_version'] = 'Unable to retrieve'
+                current_app_version = 'Unable to retrieve'
+                return current_app_version
 
     else:
-        context['release_version'] = '0.0.1'
+        current_app_version = 'Unable to retrieve metadata file'
+
+    return current_app_version
+
+def check_app_updates(request):
+    template = 'common/check_app_updates.html'
+    context = dict()
+    current_version = current_app_version()
+    url = 'https://raw.githubusercontent.com/krishnakuruvadi/portfoliomanager/main/src/metadata.json'
+
+    try:
+        request = requests.get(url)
```

Comment:
pls add a timeout for the get request.  sometimes the get operation just gets stuck for a long time

Reply:
Time-out added and recommended exceptions are now caught individually to help troubleshoot later if necessary.